src/DynamoPackagesUI/Utilities/PackageManagerCommands.cs

```diff
@@ -1,503 +1,96 @@
-                Application.Current.Dispatcher.Invoke((Action)(() =>
-                {
-                    localPkg.UninstallCore(dynModel.CustomNodeManager, Loader, dynModel.PreferenceSettings);
-                }));
-
-                return true;
-            }
-            catch (Exception)
-            {
-                MessageBox.Show(string.Format(Resources.MessageFailedToUninstall,
-                    ProductName),
-                    Resources.UninstallFailureMessageBoxTitle,
-                    MessageBoxButton.OK, MessageBoxImage.Error);
-                return false;
+                case PackageManagerMessages.FAIL_TO_UNINSTALL_PACKAGE: return string.Format(Resources.MessageFailToUninstallPackage, args);
+                case PackageManagerMessages.CONFIRM_TO_INSTALL_PACKAGE: return string.Format(Resources.MessageConfirmToInstallPackage, args);
+                case PackageManagerMessages.UNINSTALL_TO_CONTINUE: return string.Format(Resources.MessageUninstallToContinue, args);
+                case PackageManagerMessages.UNINSTALL_TO_CONTINUE2: return string.Format(Resources.MessageUninstallToContinue2, args);
+                case PackageManagerMessages.ALREADY_INSTALLED_DYNAMO: return string.Format(Resources.MessageAlreadyInstallDynamo, args);
+                case PackageManagerMessages.PACKAGE_CONTAIN_PYTHON_SCRIPT: return string.Format(Resources.MessagePackageContainPythonScript, args);
+                case PackageManagerMessages.PACKAGE_NEWER_DYNAMO: return string.Format(Resources.MessagePackageNewerDynamo, args);
+                case PackageManagerMessages.NEED_TO_RESTART: return string.Format(Resources.MessageNeedToRestart, args);
+                case PackageManagerMessages.CONFIRM_TO_UNINSTALL: return string.Format(Resources.MessageConfirmToUninstallPackage, args);
+                case PackageManagerMessages.FAILED_TO_UNINSTALL: return string.Format(Resources.MessageFailedToUninstall, args);
+                case PackageManagerMessages.CONFIRM_TO_INSTALL_PACKAGE_FOLDER: return string.Format(Resources.MessageConfirmToInstallPackageToFolder, args);
             }
+            return string.Empty;
         }
 
-        /// <summary>
-        /// Open the root directory for PkgRequest
-        /// </summary>
-        public void GoToRootDirectory()
-        {
-            Package localPkg = Loader.LocalPackages.Where(a => a.Name == this.PkgRequest.asset_name.ToString()).First();
-            Process.Start(localPkg.RootDirectory);
-        }
+    }
 
-        /// <summary>
-        /// Unmark Uninstall for PkgRequest 
-        /// </summary>
-        public void UnmarkForUninstallation()
-        {
-            Package pkg = Loader.LocalPackages.Where(a => a.Name == this.PkgRequest.asset_name.ToString()).First();
-            if (pkg != null)
-            {
-                pkg.UnmarkForUninstall(Model.PreferenceSettings);
-            }
-        }
+    public enum PackageManagerMessages
```

Comment:
enum name could be MessageTypes, also use camel casing for enum values, FailToUninstallPackage instead of FAIL_TO_UNINSTALL_PACKAGE.